### DISCUSSION TITLE
Search with geojson=1 302 redirects to request with jsonv2=1

### DISCUSSION THREAD

**Discussion by mike-little** [osm-search/Nominatim#3461] [Q&A] [answered]
Hi,

Using java the API, a call using this URL, where I requested a GeoJson(geojson=1) response resulted in a 302 redirect with the new URL requesting jsonv2(jsonv2=1), the address was found through the redirect and therefore I received jsonv2 instead of GeoJson.  Is this expected and must be handled by callers, any documentation available on this if so?  Additionally it seems to lose my request for namedetails=1 and extratags=1.   Below is the original request along with the redirect URL I see in the browser.

Maybe related to the address number, #209, as I don't see that included in the browser details even though that is what I entered in my request, maybe it truncated the request, even so, odd that it did the redirect?

https://nominatim.openstreetmap.org/search?q=40%20W%20Littleton%20Blvd%20#209,%20Littleton,%20CO%2080120&format=geojson&addressdetails=1&extratags=1&namedetails=1&limit=10

https://nominatim.openstreetmap.org/ui/search.html?q=40%20W%20Littleton%20Blvd%20

Then UI sends this request:

https://nominatim.openstreetmap.org/search.php?q=40%20W%20Littleton%20Blvd%20&polygon_geojson=1&format=jsonv2

**Comment by rscop** [comment]
I think your problem is the house number (Im assuming # 209 is it), because if you search without it the API will return

https://nominatim.openstreetmap.org/search?q=40%20W%20Littleton%20Blvd%20Littleton,%20CO%2080120&format=geojson&addressdetails=1&extratags=1&namedetails=1&limit=10

> I received jsonv2 instead of GeoJson. Is this expected and must be handled by callers, any documentation available on this if so? Additionally it seems to lose my request for namedetails=1 and extratags=1

The jsonv2 is the [default format](https://nominatim.org/release-docs/latest/api/Search/#output-format) for output response if you have not defined any.
The request is returning a geojson for the point, and also extrags and namedetails, but they are null.

**Reply by mtmail** [reply, marked as answer]
Yes, a `#` in a URL starta s URL fragment. The webserver only looks at the part before `#` to create the redirect. The `#` needs to be escaped (into `%23`)

**Reply by mike-little** [reply]
Thanks for the quick replies here.   Yeah, encoding the hash allows me to get further.   So, now the full search text is apparently interpreted in it's entirety.     I now get a geojson response for my request that has format=geojson.   

https://nominatim.openstreetmap.org/search?q=40%20W%20Littleton%20Blvd%20#209,%20Littleton,%20CO%2080120&format=geojson&addressdetails=1&extratags=1&namedetails=1&limit=10

However, I get no results back, just an empty list of features.

Generally, I am finding address searches for multi-unit locations where a suite, apt, etc is included in the free form search text results in an empty response.  Examples:   "26 W Dry Creek Cir #300, Littleton, CO 80120", "1934 Old Gallows Rd STE 350, Tysons, VA 22182".

If I remove the multi-unit "unit #" from the search and search for the building address I get results.   Examples: "26 W Dry Creek Cir, Littleton, CO 80120",  "1934 Old Gallows Rd, Tysons, VA 22182".

Could the algorithm be improved to find the lat/lng/location info in these cases by stripping out the unit # seamlessly?

**Reply by mtmail** [reply]
please open a new discussion
